drivers/watchdog/wdt_nrfx.c

```diff
@@ -200,7 +201,25 @@ static const struct wdt_driver_api wdt_nrf_api = {
 
 static int init_wdt(struct device *dev)
 {
+	nrfx_err_t err_code;
+	nrfx_wdt_config_t config;
+
 	ARG_UNUSED(dev);
+
+	/* Set default values. They will be overwritten by setup function. */
+	config.behaviour = NRF_WDT_BEHAVIOUR_RUN_SLEEP_HALT;
+	config.reload_value = 2000;
+
+	err_code = nrfx_wdt_init(&config, wdt_event_handler);
+	if (err_code != NRFX_SUCCESS) {
+		return -EBUSY;
+	}
+
+	IRQ_CONNECT(CONFIG_WDT_NRF_IRQ,
+		    CONFIG_WDT_NRF_IRQ_PRI, wdt_nrf_isr,
+		    DEVICE_GET(wdt_nrfx), 0);
```

Comment:
Instead of `wdt_nrf_isr, DEVICE_GET(wdt_nrfx)`, use `nrfx_isr, nrfx_wdt_irq_handler`.